cpp/autosar/src/rules/M0-1-4/SingleUsePODVariable.qll

```diff
@@ -10,23 +10,69 @@ private string getConstExprValue(Variable v) {
   v.isConstexpr()
 }
 
+/**
+ * Gets the number of uses of variable `v` in an opaque assignment, where an opaqua assignment for example a cast from one type to the other and `v` is assumed to be a member of the resulting type.
+ * e.g.,
+ * struct foo {
+ *  int bar;
+ * }
+ *
+ * struct foo * v = (struct foo*)buffer;
+ */
+Expr getIndirectSubObjectAssignedValue(MemberVariable subobject) {
```

Comment:
without looking at this too closely (but just to double check)... is there anything in the existing [aggregate literal initializers concepts](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/enhancements/AggregateLiteralEnhancements.qll) already that can be reused here?

Reply:
No, this looks at cases where you for example cast a byte buffer to a struct. This reinterpretation "assigns" a value to a field of struct, without actually assigning it. However, we still want to count that as a use to ensure we alert on those fields.